src/glob.js

```diff
@@ -1,13 +1,97 @@
+import fs from "node:fs";
+import path from "node:path";
 import { glob } from "glob";
+import ignore from "ignore";
 import logger from "./logger.js";
 
-async function getFiles(pattern) {
+class NotFound extends Error {}
+
+async function getMatches(pattern) {
   try {
-    return await glob(pattern, { dot: true, dotRelative: true });
+    return await glob(pattern, { dot: true });
   } catch (e) {
     logger.error(e.message);
     return [];
   }
 }
 
-export { getFiles };
+async function exists(path) {
+  try {
+    await fs.promises.access(path);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function filterIgnores(filenames, ignorePatterns) {
+  const ig = ignore();
+  for (const patterns of ignorePatterns) {
+    ig.add(patterns);
+  }
+  return ig.filter(filenames);
+}
+
+async function readIgnoreFiles(filenames) {
+  let content = [];
+  for (const filename of filenames) {
+    const abspath = path.join(process.cwd(), filename);
+    if (await exists(abspath)) {
+      content.push(await fs.promises.readFile(abspath, "utf8"));
+    }
+  }
+  return content;
+}
+
+function getSeparator() {
+  const platform =
+    typeof process === "object" &&
+    process &&
+    typeof process.platform === "string"
+      ? process.platform
+      : "linux";
+
+  return platform == "win32" ? "\\" : "/";
+}
```

Comment:
TODO: Can I just assume `process.platform` exists given this is node only and won't be run in a browser?